build-logic/build-update-utils/src/main/kotlin/gradlebuild/buildutils/tasks/AbstractCheckOrUpdateContributorsInReleaseNotes.kt

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild.buildutils.tasks
+
+import com.google.gson.Gson
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.options.Option
+import org.gradle.work.DisableCachingByDefault
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+
+
+data class GitHubPullRequestSearchResult(val items: List<GitHubPullRequest>)
+
+
+data class GitHubUser(val login: String, val name: String?)
+
+
+data class GitHubPullRequestMilestone(val title: String)
+
+
+data class GitHubPullRequest(
+    val number: Int,
+    val user: GitHubUser,
+    val milestone: GitHubPullRequestMilestone?,
+)
+
+
+val contributorSectionRegex =
+    "(?s)(.*We would like to thank the following community members for their contributions to this release of Gradle:\\n)(.*)(\\n<!--\nInclude only their name.*)".toRegex()
+
+
+val contributorLineRegex = "\\[(.*)]\\(https://github.com/(.*)\\)".toRegex()
+
+
+const val pageSize = 100
+
+
+@DisableCachingByDefault(because = "Depends on GitHub API")
+abstract class AbstractCheckOrUpdateContributorsInReleaseNotes : DefaultTask() {
+    @get: Internal
+    abstract val releaseNotes: RegularFileProperty
+
+    @get: Option(option = "milestone", description = "The milestone prefix to check for contributors, i.e. '7.5' includes PRs with '7.5', '7.5 RC1', '7.5.1'")
+    @get: Internal
+    abstract val milestone: Property<String>
+
+    @get: Option(option = "github-token", description = "The GitHub token to use for API requests. Will use anonymous requests if not specified.")
```

Comment:
We should make this pull from somewhere else and not the command-line since this will be captured in build scans.